google/cloud/bigtable/mutations.py

```diff
@@ -13,42 +13,121 @@
-    pass
+    def is_idempotent(self) -> bool:
+        """
+        Check if the mutation is idempotent
+        If false, the mutation will not be retried
+        """
+        return True
+
+    def __str__(self) -> str:
+        return str(self._to_dict())
 
 
 @dataclass
 class SetCell(Mutation):
-    family: family_id
-    column_qualifier: qualifier
-    new_value: bytes | str | int
-    timestamp_ms: int | None = None
+    family: str
+    qualifier: bytes
+    new_value: bytes
+    timestamp_micros: int | None = None
+
+    def _to_dict(self) -> dict[str, Any]:
+        """Convert the mutation to a dictionary representation"""
+        # if timestamp not given, use -1 for server-side timestamp
+        timestamp = self.timestamp_micros if self.timestamp_micros is not None else -1
+        return {
+            "set_cell": {
+                "family_name": self.family,
+                "column_qualifier": self.qualifier,
+                "timestamp_micros": timestamp,
+                "value": self.new_value,
+            }
+        }
+
+    def is_idempotent(self) -> bool:
+        """Check if the mutation is idempotent"""
+        return self.timestamp_micros is not None and self.timestamp_micros >= 0
 
 
 @dataclass
 class DeleteRangeFromColumn(Mutation):
-    family: family_id
-    column_qualifier: qualifier
-    start_timestamp_ms: int
-    end_timestamp_ms: int
+    family: str
+    qualifier: bytes
+    # None represents 0
+    start_timestamp_micros: int | None = None
+    # None represents infinity
+    end_timestamp_micros: int | None = None
+
+    def __post_init__(self):
+        if (
+            self.start_timestamp_micros is not None
+            and self.end_timestamp_micros is not None
+            and self.start_timestamp_micros > self.end_timestamp_micros
+        ):
+            raise ValueError("start_timestamp_micros must be <= end_timestamp_micros")
+
+    def _to_dict(self) -> dict[str, Any]:
+        timestamp_range = {}
+        if self.start_timestamp_micros is not None:
+            timestamp_range["start_timestamp_micros"] = self.start_timestamp_micros
+        if self.end_timestamp_micros is not None:
+            timestamp_range["end_timestamp_micros"] = self.end_timestamp_micros
+        return {
+            "delete_from_column": {
+                "family_name": self.family,
+                "column_qualifier": self.qualifier,
+                "time_range": timestamp_range,
+            }
+        }
 
 
 @dataclass
 class DeleteAllFromFamily(Mutation):
-    family_to_delete: family_id
+    family_to_delete: str
+
+    def _to_dict(self) -> dict[str, Any]:
+        return {
+            "delete_from_family": {
+                "family_name": self.family_to_delete,
+            }
+        }
 
 
 @dataclass
 class DeleteAllFromRow(Mutation):
-    pass
+    def _to_dict(self) -> dict[str, Any]:
+        return {
+            "delete_from_row": {},
+        }
 
 
-@dataclass
 class BulkMutationsEntry:
```

Comment:
Can we rename this to RowMutationEntry? because all the mutations are targeting a single row https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/RowMutationEntry.java#L26-L32

Reply:
My only concern is that it gets a bit confusing with `mutate_row` and `bulk_mutate_rows`. I'm only using this class for the `bulk_mutate_rows`, which is why I added the Bulk

But I'm fine with the change, especially if that's how Java does it